homeassistant/components/comelit/config_flow.py

```diff
@@ -0,0 +1,174 @@
+"""Config flow for Comelit integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+import aiocomelit
+import voluptuous as vol
+
+from homeassistant import core, exceptions
+from homeassistant.components.hassio import HassioServiceInfo
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PIN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import _LOGGER, DEFAULT_HOST, DEFAULT_PIN, DOMAIN
+from .coordinator import ComelitSerialBridge
+
+
+def user_form_schema(user_input: dict[str, Any] | None) -> vol.Schema:
+    """Return user form schema."""
+    user_input = user_input or {}
+    return vol.Schema(
+        {
+            vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+            vol.Optional(CONF_PIN, default=DEFAULT_PIN): str,
+        }
+    )
+
+
+STEP_REAUTH_DATA_SCHEMA = vol.Schema({vol.Required(CONF_PIN): str})
+
+
+async def validate_input(
+    hass: core.HomeAssistant, data: dict[str, Any]
+) -> dict[str, str]:
+    """Validate the user input allows us to connect."""
+
+    coordinator = ComelitSerialBridge(hass, data[CONF_HOST], data[CONF_PIN])
```

Comment:
I think it is not needed to initialize the whole coordinator with all it's overhead, when only the api connection should be tested
```suggestion 
    api = ComeliteSerialBridgeAPi(data[CONF_HOST], data[CONF_PIN])
```

Reply:
Done